src/dotnet/Telemetry.cs

```diff
@@ -146,5 +155,49 @@ private void TrackEventTask(string eventName, IDictionary<string, string> proper
                 return _commonProperties;
             }
         }
+        
+        private List<string> GetMacs()
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            var macs = new List<string>();
+            
+            if (nics == null || nics.Length < 1)
+            {
+                macs.Add(string.Empty);
+                return macs;
+            }
+            
+            foreach (NetworkInterface adapter in nics)
+            {
+                IPInterfaceProperties properties = adapter.GetIPProperties();
+
+                PhysicalAddress address = adapter.GetPhysicalAddress();
+                byte[] bytes = address.GetAddressBytes();
+                macs.Add(string.Join("-", bytes.Select(x => x.ToString("X2"))));
```

Comment:
For future correlation purposes(e.g. matching the hashed value with VsCode's telemetry). I have added a comment in the code to indicate the need for a specific format.